jwql/jwql_monitors/monitor_cron_jobs.py

```diff
@@ -0,0 +1,311 @@
+
+    stdev_delta : float
+        Standard deviation in seconds between the appearance of consecutive log files
+    """
+    minimum_log_num = 3  # Set to a low value for now since we don't have many logfiles
+    times = [os.path.getctime(filename) for filename in filenames]
+    if len(times) >= minimum_log_num:
+        sorted_times = np.array(sorted(times))
+        delta_times = sorted_times[1:] - sorted_times[0:-1]
+        mean_delta = np.mean(delta_times)
+        stdev_delta = np.std(delta_times)
+    else:
+        # If there are < minimum_log_num logfiles, then let's assume we can't
+        # get a reliable measure of cadence. Fall back to a value of
+        # 1 year between files, to avoid accidentally flagging this monitor
+        # as running late in the subsequent check
+        mean_delta = 31556736.0  # sec per year
+        stdev_delta = 31556736.0  # sec per year
+    return mean_delta, stdev_delta
+
+
+@log_fail
+@log_info
+def status(production_mode=True):
+    """Main function: determine the status of the instrument montiors by examining
+    log files.
+
+    Parameters
+    ----------
+    production_mode : bool
+        If true, look in the main log directory. If false, look in the dev log
+        file directory.
+
+    Returns
+    -------
+    logfile_status : dict
+        Nested dictionary containing the status for all monitors. Top level keys
+        include all monitors. Within a given monitor, the value is a dictionary
+        containing 'missing_file' and 'status' keys. 'missing_file' is a boolean
+        describing whether or not there is a suspected missing log file based
+        on the timestamps of the existing files. 'status' is a string that is
+        either 'success' or 'failure'.
+    """
+    # Begin logging
+    configure_logging('monitor_cron_jobs', production_mode=production_mode)
+    logging.info("Beginning cron job status monitor")
+
+    # Get main logfile path
+    settings = get_config()
+    log_path = settings['log_dir']
+
+    # If we are in development mode, the log files are in a slightly
+    # different location than in production mode
+    if not production_mode:
+        log_path = os.path.join(log_path, 'dev')
+
+    # Set up a dictionary to keep track of results
+    logfile_status = {}
+
+    # Get a list of the directories under the main logging directory.
+    generator = os.walk(log_path, topdown=True)
+
+    # Loop over monitors (note that subsubdir should always be empty)
+    for subdir, subsubdir, filenames in generator:
+        if len(filenames) > 0:
+            monitor_name = subdir.split('/')[-1]
+            log_file_list = [os.path.join(subdir, filename) for filename in filenames]
+
+            # Find the cadence of the monitor
+            delta_time, stdev_time = get_cadence(log_file_list)
+
+            # Identify the most recent log file
+            latest_log, latest_log_time = find_latest(log_file_list)
```

Comment:
@bhilbert4 If you do `log_file_list = sorted(log_file_list)` then the `latest_log` should be as trivial as grabbing `log_file_list[-1]`.  Sorting it this way may also make `get_cadence()` a bit easier ?

Reply:
Interesting. I hadn't thought about that. Is it guaranteed that log file names will be such that `sorted` will always work? I guess the way `make_log_file` is currently set up does guarantee that.

I replaced the line in `find_latest` that was using the timestamps with your fix above.

Reply:
I just did some timing tests and obtaining and sorting the timestamps of all log files is comparable to converting the time strings from the filenames to datetime objects and sorting those.

Reply:
Some better timing tests are showing the use of timestamps is actually just slightly faster than going by the time information in the filenames.